StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/Link/ConsumerSession.swift

```diff
@@ -12,81 +12,50 @@ import UIKit
 @_spi(STP) import StripePayments
 
 /// For internal SDK use only
-@objc(STP_Internal_ConsumerSession)
-class ConsumerSession: NSObject, STPAPIResponseDecodable {
-
+final class ConsumerSession: Decodable {
     let clientSecret: String
     let emailAddress: String
     let redactedPhoneNumber: String
     let verificationSessions: [VerificationSession]
-    let authSessionClientSecret: String?
```

Comment:
Hoisting this out into a separate response type (see [`SessionResponse` in STPAPIClient+Link.swift](https://github.com/stripe/stripe-ios/pull/2291/files#diff-37b1d3022a0265c5ae1a48d7f4f2027426df1271f4c2b03bacb2ce6bc5b7e3b4)) was informed by a couple of factors, but one was the lack of support in `StripeJSONDecoder` for nested containers - using a `nestedKeyedContainer` was the most sensible way to implement `init(from:)` here while still parsing `authSessionClientSecret` correctly, but that wasn't viable. A previous implementation for this PR actually added support for them to enable this, but looking over the [original PR](https://github.com/stripe-ios/stripe-ios-private/pull/979), it seems like some of these limitations may have actually been intentional, so I took another approach. I'm happier with this one overall—smaller surface area, better matching of object shapes to API shapes—but @davidme-stripe would love to get some insight around this!

Reply:
This approach seems good to me! Re: Nested container support, I probably shouldn't have said *intentionally* in the original PR — I was just noting that it wasn't an oversight. We didn't have any plans to use it, so I skipped it in the interest of time.

Reply:
Gotcha! Happy to tackle adding it back in at some future date, but feels a touch too scope-creepy for this PR.